tensorflow_lite_support/cc/task/vision/utils/image_tensor_specs.cc

```diff
@@ -177,13 +177,32 @@ StatusOr<ImageTensorSpecs> BuildInputImageTensorSpecs(
 
   // The expected layout is BHWD, i.e. batch x height x width x color
   // See https://www.tensorflow.org/guide/tensors
-  int batch = input_tensor->dims->data[0];
+  const int batch = input_tensor->dims->data[0];
   int height = input_tensor->dims->data[1];
   int width = input_tensor->dims->data[2];
-  int depth = input_tensor->dims->data[3];
+  const int depth = input_tensor->dims->data[3];
   // Determine if the input shape is resizable.
   const TfLiteIntArray* dims_signature = input_tensor->dims_signature;
 
+  if (dims_signature != nullptr)
+  {
+      if (dims_signature->data[3] == -1 || dims_signature->data[0] == -1)
+        return CreateStatusWithPayload(StatusCode::kInvalidArgument,
+                                       "Only height and width is mutable for now.",
+                                       TfLiteSupportStatus::kInvalidArgumentError);
+      
+      const int height_mutable = dims_signature->data[1];
+      const int width_mutable = dims_signature->data[2];
+      
+      // How do I get the image height and width???
```

Comment:
Height and width won't be populated here, because FrameBuffer is passed in through [BaseVisionTaskApi::Preprocess](https://github.com/tensorflow/tflite-support/blob/ba16516f0f06558a38201958253b42e14c913447/tensorflow_lite_support/cc/task/vision/core/base_vision_task_api.h#L115) or [ImagePreprocessor::Preprocess](https://github.com/tensorflow/tflite-support/blob/f52f2d9b121aae7754eb512006a8734c113d7175/tensorflow_lite_support/cc/task/processor/image_preprocessor.h#L76). 

After a second thought, let's make changes directly to ImageProcessor instead of BaseVisionTaskApi, so that we don't need to move the changes from BaseVisionTaskApi to ImageProcessor later. What do you think?

Reply:
Okay but does it have access to the interpreter (to resize the graph and to access input tensor)?

Reply:
TfLiteEngine is a member of the base classes, Preprocessor and Postprocessor. See the code here: https://github.com/tensorflow/tflite-support/blob/f52f2d9b121aae7754eb512006a8734c113d7175/tensorflow_lite_support/cc/task/processor/processor.h#L106

Reply:
Ah , the engine was the wrapper to the interpreter if I recall correctly. So i could use ```engine_``` for resize operator. Correct?

Reply:
That's right. You can retrieve Interpreter from tflite_engine.